text/0000-contingency-plan.md

```diff
@@ -0,0 +1,425 @@
+disruptive to the ecosystem. Therefore, **the RFC also proposes
+specific measures to mitigate the impact of breaking changes**, and
+some criteria when those measures might be appropriate.
+
+Furthermore, there are other kinds of changes that we may want to make
+which feel like they *ought* to be possible, but which are in fact
+breaking changes. The simplest example is adding a new keyword to the
+language -- despite being a purely additive change, a new keyword can
+of course conflict with existing identifiers. Therefore, **the RFC
+proposes a simple annotation that allows crates to designate the
+version of the language they were written for**. This effectively
+permits some amount of breaking changes by making them "opt-in"
+through the version attribute.
+
+However, even though the version attribute can be used to make
+breaking changes "opt-in" (and hence not really breaking), this is
+still a tool to be used with great caution. Therefore, **the RFC also
+proposes guidelines on when it is appropriate to include an "opt-in"
+breaking change and when it is not**.
+
+This RFC is focused specifically on the question of what kinds of
+changes we can make in a minor release (as well as some limited
+mechanisms that lay the groundwork for certain kinds of anticipated
+changes). It intentionally does not address the question of a release
+schedule for Rust 2.0, nor does it propose any new features
+itself. These topics are complex enough to be worth considering in
+separate RFCs.
+
+# Detailed design
+
+The detailed design is broken into two major section: how to address
+soundness changes, and how to address other, opt-in style changes. We
+do not discuss non-breaking changes here, since obviously those are
+safe.
+
+### Soundness changes
+
+When compiler bugs or soundness problems are encountered in the
+language itself (as opposed to in a library), clearly they ought to be
+fixed. However, it is important to fix them in such a way as to
+minimize the impact on the ecosystem.
+
+The first step then is to evaluate the impact of the fix on the crates
+found in the `crates.io` website (using e.g. the crater tool). If
+impact is found to be "small" (which this RFC does not attempt to
+precisely define), then the fix can simply be landed. As today, the
+commit message of any breaking change should include the term
+`[breaking-change]` along with a description of how to resolve the
+problem, which helps those people who are affected to migrate their
+code. A description of the problem should also appear in the relevant
+subteam report.
+
+In cases where the impact seems larger, the following steps can be
+taken to ease the transition:
+
+1. Identify important crates (such as those with many dependencies)
+   and work with the crate author to correct the code as quickly as
+   possible, ideally before the fix even lands.
+2. Work hard to ensure that the error message identifies the problem
+   clearly and suggests the appropriate solution.
```

Comment:
This assumes that the fix is to introduce an error rather than a silent change in behaviour. That is obviously preferable, but perhaps there will be situations where it is not possible. How will we handle those?